cmake/modules/armadillo.cmake

```
@@ -3,13 +3,19 @@ option(ENABLE_ARMADILLO "Add armadillo" OFF)
 if(ENABLE_ARMADILLO)
   message(STATUS "Armadillo enabled")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DMARDYN_ARMADILLO")
-  find_library(BLAS_LIB blas)
-  find_library(LAPACK_LIB lapack)
-  if(BLAS_LIB AND LAPACK_LIB)
-      message(STATUS "libblas and liblapack found")
+  find_package(BLAS)
+  find_package(LAPACK)
```

you can (and probably should) mark the packages as "required" here. Not sure about the syntax though...
```suggestion
  find_package(BLAS REQUIRED)
  find_package(LAPACK REQUIRED)
```
Then you can also throw away the if / else.